curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -827,13 +838,35 @@ void FileCacheManager::ReleaseCache() {
     auto iter = chunkCacheMap_.begin();
 
     for (; iter != chunkCacheMap_.end(); iter++) {
-        iter->second->ReleaseCache(s3ClientAdaptor_);
+        iter->second->ReleaseCache();
     }
 
     chunkCacheMap_.clear();
     return;
 }
 
+void FileCacheManager::TruncateCache(uint64_t offset, uint64_t fileSize) {
+    uint64_t chunkSize = s3ClientAdaptor_->GetChunkSize();
+    uint64_t chunkIndex = offset / chunkSize;
+    uint64_t chunkPos = offset % chunkSize;
+    int chunkLen = 0;
+    while (fileSize > 0) {
+        if (chunkPos + fileSize > chunkSize) {
+            chunkLen = chunkSize - chunkPos;
+        } else {
+            chunkLen = fileSize;
+        }
+        ChunkCacheManagerPtr chunkCacheManager =
+            FindOrCreateChunkCacheManager(chunkIndex);
+        chunkCacheManager->TruncateCache(chunkPos);
+        fileSize -= chunkLen;
```

Comment:
Why TruncateCache One by One, the  hole chunk of  (offset / chunkSize+1,  fileSize/chunkSize) should be throw